src/test/regress/sql/with.sql

```diff
@@ -1065,3 +1065,39 @@ WITH cte1 AS (
 SELECT * FROM cte1 a JOIN cte1 b USING (c1);
 
 DROP TABLE t1;
+
+-- Test join prefetch_inner in the case of bottleneck join
+SET optimizer = off;
+--start_ignore
+DROP TABLE IF EXISTS d;
+--end_ignore
+CREATE TABLE d (c1 int, c2 int) DISTRIBUTED BY (c1);
+
+INSERT INTO d (VALUES ( 2, 0 ),( 2 , 0 ));
+
+-- Test with HashJoin
+EXPLAIN (ANALYZE, COSTS OFF) WITH cte AS (
+	SELECT count(*) c1 FROM d
+) SELECT * FROM cte a JOIN (SELECT * FROM d JOIN cte USING (c1) LIMIT 1) b USING (c1);
+
+SET enable_hashjoin = off;
+SET enable_nestloop = on;
+
+-- Test with nestedloop join
```

Comment:
What exactly is this case testing if it produces the same result even without a patch?

Reply:
If we are testing prefetch we should leave outer part empty to see that inner part is ( never executed) for Nest Loop and Merge Join. And is executed for HashJoin

Otherwise there is no sense to even test the changes